Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,562 @@
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+The `options.Filter` property is an instance of `FilterQueryOption`. It will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$filter=Id eq 1
+```
+
+If we don't want to pass the filter query option in the request, but we want a filter to be applied, we can initialize the `FilterQueryOption` and call `ApplyTo` to apply the filter query to the data.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var filter = new FilterQueryOption("Id eq 1", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$filter", "Id eq 1" } },
+            options.Context.RequestContainer));
+
+    IQueryable results = filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+When making the request, we don't need to use the `$filter` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+Below will be the output:
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31
+        }
+    ]
+}
+```
+
+### Apply query
+Apply query feature is applied in grouping and aggregating data.
+
+We call `ApplyTo` on the `Apply` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Apply.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+The `options.Apply` property is an instance of `ApplyQueryOption`. It will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$apply=aggregate(Age with max as MaxAge)
+```
+
+If we don't want to pass the query option in the request, but we want to aggregate the results, we can initialize the `ApplyQueryOption` and set it in the request.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var queryOptionParser = new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$apply", "aggregate(Age with max as MaxAge)" } },
+            options.Context.RequestContainer);
+
+    options.Request.ODataFeature().ApplyClause = new ApplyQueryOption("aggregate(Age with max as MaxAge)", options.Context, queryOptionParser).ApplyClause;
```

Comment:
Not a deal breaker but the 3 code snippets provided for initializing and setting the query option on the request directly are not uniform... This one has a `queryOptionParser` variable... Then for the filter query, you don't set `FilterClause` while for `SelectExpand` and `Apply` queries you set `SelectExpandClause` and `ApplyClause` respectfully. Making them uniform could probably make it easy to relate one with the other in a small way.

Reply:
ODataFeature only has `SelectExpandClause` and `ApplyClause` but no `FilterClause`.